tests/utils/utils.py

```diff
@@ -0,0 +1,126 @@
+import json
+import logging
+from contextlib import contextmanager
+from pathlib import Path
+from time import perf_counter
+from typing import Union
+
+import pandas as pd
+from starkware.starknet.testing.starknet import StarknetContract
+
+logging.basicConfig(format="%(levelname)-8s %(message)s")
+logger = logging.getLogger("timer")
+
+
+_time_report = []
+_resources_report = []
+
+
+def timeit(fun):
+    async def timed_fun(*args, **kwargs):
+        start = perf_counter()
+        res = await fun(*args, **kwargs)
+        stop = perf_counter()
+        duration = stop - start
+        _time_report.append(
+            {"name": fun.__name__, "args": args, "kwargs": kwargs, "duration": duration}
+        )
+        logger.info(
+            f"{fun.__name__}({json.dumps(args)}, {json.dumps(kwargs)}) in {duration:.2f}s"
+        )
+        return res
+
+    return timed_fun
+
+
+class traceit:
+
+    _context = ""
+
+    @classmethod
+    @property
+    def prefix(cls):
+        return cls._context + ":" if cls._context != "" else ""
```

Comment:
pardon my python ignorance but what does this do?

Reply:
you mean what a `@property` is ? high level answer is: define something that looks like an attribute but is computed